raft/src/test/java/org/apache/kafka/raft/QuorumStateTest.java

```diff
@@ -945,11 +946,10 @@ public void testObserverUnattachedToFollower() throws IOException {
     }
 
     @Test
-    public void testInitializeWithCorruptedStore() throws IOException {
-        QuorumStateStore stateStore = Mockito.mock(QuorumStateStore.class);
-        Mockito.doThrow(IOException.class).when(stateStore).readElectionState();
+    public void testInitializeWithCorruptedStore() {
         QuorumState state = buildQuorumState(Utils.mkSet(localId));
-
+        QuorumStateStore stateStore = Mockito.mock(QuorumStateStore.class);
+        Mockito.doThrow(UncheckedIOException.class).when(stateStore).readElectionState();
```

Comment:
`nit`: I think it'd better to put the 2 line mock at the beginning of the test as before. (i.e. before `QuorumState state = buildQuorumState(Utils.mkSet(localId));`)

Reply:
@showuon Please review the changes again! Thanks.